<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Daniel Martin, Kameron Standhaft, Esra Ari Acar">
<meta name="dcterms.date" content="2023-04-18">

<title>Latent Class Analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="index_files/libs/clipboard/clipboard.min.js"></script>
<script src="index_files/libs/quarto-html/quarto.js"></script>
<script src="index_files/libs/quarto-html/popper.min.js"></script>
<script src="index_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="index_files/libs/quarto-html/anchor.min.js"></script>
<link href="index_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="index_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="index_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="index_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="index_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

<script src="index_files/libs/kePrint-0.0.1/kePrint.js"></script>
<link href="index_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="styles.css">
</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">1. Introduction</a>
  <ul class="collapse">
  <li><a href="#lca-applications" id="toc-lca-applications" class="nav-link" data-scroll-target="#lca-applications">1.1 LCA Applications</a></li>
  <li><a href="#challenges-with-implementation-and-analysis-of-classes-using-lca" id="toc-challenges-with-implementation-and-analysis-of-classes-using-lca" class="nav-link" data-scroll-target="#challenges-with-implementation-and-analysis-of-classes-using-lca">1.2 Challenges with Implementation and Analysis of Classes using LCA</a></li>
  </ul></li>
  <li><a href="#methods" id="toc-methods" class="nav-link" data-scroll-target="#methods">2. Methods</a>
  <ul class="collapse">
  <li><a href="#explanation-of-latent-class-analysis-lca" id="toc-explanation-of-latent-class-analysis-lca" class="nav-link" data-scroll-target="#explanation-of-latent-class-analysis-lca">2.1 Explanation of Latent Class Analysis (LCA)</a></li>
  <li><a href="#general-steps-of-latent-class-analysis-lca" id="toc-general-steps-of-latent-class-analysis-lca" class="nav-link" data-scroll-target="#general-steps-of-latent-class-analysis-lca">2.2 General Steps of Latent Class Analysis (LCA)</a></li>
  </ul></li>
  <li><a href="#analysis-and-results" id="toc-analysis-and-results" class="nav-link" data-scroll-target="#analysis-and-results">3. Analysis and Results</a>
  <ul class="collapse">
  <li><a href="#software" id="toc-software" class="nav-link" data-scroll-target="#software">3.1 Software</a></li>
  <li><a href="#dataset-description" id="toc-dataset-description" class="nav-link" data-scroll-target="#dataset-description">3.2 Dataset Description</a></li>
  <li><a href="#load-data" id="toc-load-data" class="nav-link" data-scroll-target="#load-data">3.3 Load Data</a></li>
  <li><a href="#data-preparation" id="toc-data-preparation" class="nav-link" data-scroll-target="#data-preparation">3.4 Data Preparation</a></li>
  </ul></li>
  <li><a href="#statistical-modeling" id="toc-statistical-modeling" class="nav-link" data-scroll-target="#statistical-modeling">4. Statistical Modeling</a>
  <ul class="collapse">
  <li><a href="#fitting-lca-models-with-2-7-of-classes" id="toc-fitting-lca-models-with-2-7-of-classes" class="nav-link" data-scroll-target="#fitting-lca-models-with-2-7-of-classes">4.1 Fitting LCA Models with 2-7 # of Classes</a></li>
  <li><a href="#model-comparison-statistics" id="toc-model-comparison-statistics" class="nav-link" data-scroll-target="#model-comparison-statistics">4.2 Model Comparison Statistics</a></li>
  <li><a href="#chosen-model-discussion-and-assessment" id="toc-chosen-model-discussion-and-assessment" class="nav-link" data-scroll-target="#chosen-model-discussion-and-assessment">4.3 Chosen Model Discussion and Assessment</a></li>
  <li><a href="#sensitivity-analysis---correlation" id="toc-sensitivity-analysis---correlation" class="nav-link" data-scroll-target="#sensitivity-analysis---correlation">4.4 Sensitivity Analysis - Correlation</a></li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion">Conclusion</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Latent Class Analysis</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Daniel Martin, Kameron Standhaft, Esra Ari Acar </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">April 18, 2023</p>
    </div>
  </div>
    
  </div>
  

</header>

<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">1. Introduction</h2>
<p>Latent Class Analysis (LCA) is a statistical modeling method that can be utilized when it is believed that there may be unobserved subgroups (classes) among the individuals within a population. To identify these unobserved groups, LCA searches for and analyzes response patterns that may exist within categorical variables reported in the dataset <span class="citation" data-cites="nylund2018ten">(<a href="#ref-nylund2018ten" role="doc-biblioref">Nylund-Gibson and Choi 2018</a>)</span> with the goal of assigning each observation into a class based on the posterior probability of certainty that an observation should belong to a particular class <span class="citation" data-cites="vermunt2010latent">(<a href="#ref-vermunt2010latent" role="doc-biblioref">Vermunt 2010</a>)</span>. Latent Class Analysis is an example of a broader class of models called finite mixture models(FFM) <span class="citation" data-cites="grimm2021model">(<a href="#ref-grimm2021model" role="doc-biblioref">Grimm, Houpt, and Rodgers 2021</a>)</span> that seek to identify and classify instances of variance within model parameters between unobserved groups within the data that have not been explicitly defined as groups by the attributes provided within the dataset itself. Ultimately, it is the “latent class prevalence” which is being estimated through a process in which the LCA function is estimating the probability that every entry being assessed from the dataset belongs within each of the classes generated by the model<span class="citation" data-cites="law2016primer">(<a href="#ref-law2016primer" role="doc-biblioref">Law and Harrington 2016</a>)</span>.</p>
<p>LCA specifically makes the assumption that within the population potentially exists a finite number of unobserved subgroups, or latent classes, and that each individual belongs to only one of these classes <span class="citation" data-cites="weller2020latent">(<a href="#ref-weller2020latent" role="doc-biblioref">Weller, Bowen, and Faubert 2020</a>)</span>. The overarching goal of LCA is to identify where a specific subset of classes exist within a group and estimate the proportions of those subset classes within the overall population. Additionally, as shown in (Figure 1.0.1), LCA can be used to help determine if any relationships exist between the identified latent classes and any other observed variables within the dataset.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/lca_2class_overlap_example.jpg" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Figure 1.0.1 - LCA 2-Class Model Representation</figcaption><p></p>
</figure>
</div>
<p><em>The dotted lines show the 2 unobserved latent classes that exist within the overall population distribution.</em><span class="citation" data-cites="sinha2021practitioner">(<a href="#ref-sinha2021practitioner" role="doc-biblioref">Sinha, Calfee, and Delucchi 2021</a>)</span></p>
<p>As computational efficiency has improved, Latent Class Analysis has seen an uptick in interest for researchers across a variety of fields with applications in science, business, and health <span class="citation" data-cites="petersen2019application">(<a href="#ref-petersen2019application" role="doc-biblioref">Petersen, Qualter, and Humphrey 2019</a>)</span>. LCA is often most closely associated with identifying patterns of behavior or psychological characteristics that prove often otherwise difficult groups to identify<span class="citation" data-cites="kaplan2004sage">(<a href="#ref-kaplan2004sage" role="doc-biblioref">Kaplan 2004</a>)</span>.</p>
<section id="lca-applications" class="level3">
<h3 class="anchored" data-anchor-id="lca-applications">1.1 LCA Applications</h3>
<p>One frequent application of Latent Class Analysis (LCA) has been to utilize it to identify subgroups of individuals within a population that may have similar or associated symptoms or characteristics of a mental health profile <span class="citation" data-cites="petersen2019application">(<a href="#ref-petersen2019application" role="doc-biblioref">Petersen, Qualter, and Humphrey 2019</a>)</span> or when and for which specific barriers may exist to receiving medical care <span class="citation" data-cites="thorpe2011patterns">(<a href="#ref-thorpe2011patterns" role="doc-biblioref">Thorpe et al. 2011</a>)</span>. LCA has also been used to identify when any subgroups of a population exist containing individuals that may have different risk factors or potential outcomes of a particular disease or disease pattern <span class="citation" data-cites="petersen2019application">(<a href="#ref-petersen2019application" role="doc-biblioref">Petersen, Qualter, and Humphrey 2019</a>)</span>. Outside of health field applications, Latent Class Analysis can also be used in business and marketing to assess buying habits and even latent classes for groups with preferences for specific types of products <span class="citation" data-cites="kaplan2004sage">(<a href="#ref-kaplan2004sage" role="doc-biblioref">Kaplan 2004</a>)</span>.</p>
</section>
<section id="challenges-with-implementation-and-analysis-of-classes-using-lca" class="level3">
<h3 class="anchored" data-anchor-id="challenges-with-implementation-and-analysis-of-classes-using-lca">1.2 Challenges with Implementation and Analysis of Classes using LCA</h3>
<p>LCA has the ability to use mixture-modeling to distinguish and label instances of heterogeneity within a population and to identify subgroups of characteristics or individuals that may be more likely to exhibit those characteristics of heterogeneity <span class="citation" data-cites="weller2020latent">(<a href="#ref-weller2020latent" role="doc-biblioref">Weller, Bowen, and Faubert 2020</a>)</span>. Identifying and labeling subgroups of a population that may not be explicitly obvious based upon its reported attributes has substantial application for helping to identify individuals within that population who may be at greater risk or likelihood for a particular outcome. This application can help drive execution of actions that may require intervention <span class="citation" data-cites="petersen2019application">(<a href="#ref-petersen2019application" role="doc-biblioref">Petersen, Qualter, and Humphrey 2019</a>)</span>.</p>
<p>Latent Class Analysis has application across a variety of fields, but there are also limitations that exist with to its use under certain circumstances that can prove challenging to its implementation. One of the major challenges that exist in any attempted implementation of this method, is being able to determine the appropriate number of latent classes to choose for modeling the dataset. Doing so can prove time intensive and cumbersome, as the process can be somewhat subjective and often requires using a multi-step process to test multiple models and/or number of classes <span class="citation" data-cites="asparouhov2014auxiliary">(<a href="#ref-asparouhov2014auxiliary" role="doc-biblioref">Asparouhov and Muthén 2014</a>)</span>.</p>
<p>Interpretation of the classes themselves that were generated can also provide some challenge as it may require knowledge of not only the implementation methodology of the LCA upon the data but also requires that the analyst have access to background information about the data attributes and context <span class="citation" data-cites="weller2020latent">(<a href="#ref-weller2020latent" role="doc-biblioref">Weller, Bowen, and Faubert 2020</a>)</span>. Though a class may be correctly identified, if the researcher/analyst does not correctly identify what is heterogenous about that class, it can lead to improper naming and assumptions about the defining characteristics of the members of that class for analysis and application <span class="citation" data-cites="weller2020latent">(<a href="#ref-weller2020latent" role="doc-biblioref">Weller, Bowen, and Faubert 2020</a>)</span>.</p>
</section>
</section>
<section id="methods" class="level2">
<h2 class="anchored" data-anchor-id="methods">2. Methods</h2>
<section id="explanation-of-latent-class-analysis-lca" class="level3">
<h3 class="anchored" data-anchor-id="explanation-of-latent-class-analysis-lca">2.1 Explanation of Latent Class Analysis (LCA)</h3>
<p>Latent class analysis (LCA) is a statistical method used to identify hidden patterns or clusters within a dataset. This technique groups a set of observations based on several variables into distinct and exclusive categories, where the variables are unrelated within each category but similar across each one. Essentially, LCA as a type of finite mixture model (FMM), also known as unsupervised learning models, combines different distributions to group similar data based on selected parameters, resulting in data segmentation<span class="citation" data-cites="naldi2020research">(<a href="#ref-naldi2020research" role="doc-biblioref">Naldi and Cazzaniga 2020</a>)</span>.</p>
<p>In LCA, there are two types of parameters. The first type is the inclusion probabilities, which represent the probability of a random case in a population being assigned to any latent class. The second type is the conditional probabilities, which describe the likelihood of a variable taking a certain value, given that it belongs to a specific class<span class="citation" data-cites="naldi2020research">(<a href="#ref-naldi2020research" role="doc-biblioref">Naldi and Cazzaniga 2020</a>)</span>.</p>
<p>To determine the parameters of the sub-distributions in LCA, maximum likelihood estimation is often used with the expectation-maximization algorithm. This algorithm is a well-established method for evaluating the goodness of fit of a statistical model. The standard equation <span class="citation" data-cites="naldi2020research">(<a href="#ref-naldi2020research" role="doc-biblioref">Naldi and Cazzaniga 2020</a>)</span> for LCA is:</p>
<p><span class="math display">\[ p(x_i)=\sum_{k = 1}^{K}{p_k}\prod_{n = 1}^{N}{p_n(x_{in}|k)},\]</span></p>
<p>where <span class="math inline">\(p(x_i)\)</span> is the probability of observing a particular combination of responses in a group of <span class="math inline">\(N\)</span> variables, <span class="math inline">\(p_k\)</span> is the probability of membership in LC <span class="math inline">\(k\)</span>, and <span class="math inline">\(p_n(x_{in}|k)\)</span> is the probability of response to variable <span class="math inline">\(n\)</span>, conditional on membership in LC <span class="math inline">\(k\)</span>.</p>
</section>
<section id="general-steps-of-latent-class-analysis-lca" class="level3">
<h3 class="anchored" data-anchor-id="general-steps-of-latent-class-analysis-lca">2.2 General Steps of Latent Class Analysis (LCA)</h3>
<p>The general steps to use latent class analysis are as follows:</p>
<ul>
<li><p>Identify the research question and define the variables, specifically the categorical variables that will be used in the analysis</p></li>
<li><p>Select the appropriate software: In this project R Studio will be used for the LCA</p></li>
<li><p>Determine the number of latent classes to assess for ideal model</p></li>
<li><p>Estimate the model: Estimate the model parameters using maximum likelihood estimation or Bayesian estimation</p></li>
<li><p>Evaluate the model: After estimating the model, evaluate its fit using goodness-of-fit statistics such as the likelihood ratio test, Akaike Information Criterion <span class="math inline">\((AIC)\)</span>, and Bayesian Information Criterion <span class="math inline">\((BIC)\)</span>, entropy, and chi-square Goodness-of-Fit <span class="math inline">\((\chi^2)\)</span>. If the model fits well, interpret the results and report the estimated probabilities of individuals belonging to each class.</p></li>
<li><p>Conduct sensitivity analysis: Finally, conduct sensitivity analyses to test the robustness of the results and evaluate the stability of the estimated probabilities across different subgroups or samples.</p></li>
</ul>
</section>
</section>
<section id="analysis-and-results" class="level2">
<h2 class="anchored" data-anchor-id="analysis-and-results">3. Analysis and Results</h2>
<section id="software" class="level3">
<h3 class="anchored" data-anchor-id="software">3.1 Software</h3>
<p>We will be using the RStudio poLCA package library, which allows us to specify the number of classes for which we want RStudio to generate an LCA model. This package provides several tools for fitting and assessing LCA models.</p>
<p><strong>Options to be utilized:</strong></p>
<ul>
<li><strong>na.rm</strong>: allows the poLCA to function to account for any circumstances in which values may be missing within the data fields.</li>
<li><strong>graphs</strong>: displays parameter estimates as a proportion from 0.0 to 1.0, for the attributes within the model</li>
<li><strong>nrep</strong>: allows specification of the number of random starting values used by the poLCA function to assess the model for the best BIC value</li>
<li><strong>verbose</strong>: used to assess individual attribute predictions as well as concisely providing fit statistics</li>
<li><strong>maxiter</strong>: assigns the maximum number of iterations that can be used per run (<strong>nrep</strong>) to achieve convergence of the poLCA estimation algorithm</li>
</ul>
<p><strong>All Necessary Libraries are loaded here:</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#loading packages </span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggthemes)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dslabs)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(poLCA)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stats)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sjPlot)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggcorrplot)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("FactoMineR")</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("factoextra")</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(FactoMineR)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggalt)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="dataset-description" class="level3">
<h3 class="anchored" data-anchor-id="dataset-description">3.2 Dataset Description</h3>
<p>For this project we will be exploring how well Latent Class Analysis will group our dataset of zoo animals from Kaggle called <a href="https://www.kaggle.com/datasets/ulrikthygepedersen/zoo-animals">Zoo Animals</a> into subgroups based upon unmeasured homogeneity across 14 categorical variables collected for each animal.</p>
<p>There are 101 entries and a total of 18 variables in this dataset. The variables give information about the physical attributes of each of the animals, such as if they have hair, if they have teeth, if they lay eggs, etc. Of the 18 variables, 15 of them are binary variables and one of them is discrete. This majority of binary variables is why we chose this dataset for Latent Class Analysis.</p>
<p>As a comparison metric for assessing the quality of the LCA models generated, the variable that we are concerned with the most is the “type” variable. This variable has 7 categories: mammals, fish, birds, insects, reptiles, invertebrates, amphibians. Of the 101 entries, 42 are mammals (41.58%), 21 are birds (20.79%), 14 are fish (13.86%), 11 are invertebrates (10.89%), 8 are insects (7.92%), 6 are reptiles (5.94%), and 5 are amphibians (4.95%).</p>
<p>Among other variables, we will be removing the “type” variable from the dataset. Latent Class Analysis will be utilized to ascertain whether the groups identified by this method correspond to the seven animal types or if there is a different set of subclasses identified by the best LCA model as assessed by the convential LCA model assessment criteria.</p>
<p><strong>Table 3.2.1 Variables considered in LCA Analysis:</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Creating data frame of variables used and not used for displaying in tables</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>tab_lca_var <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">var_num =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">14</span>, </span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>   <span class="at">Variable_data_type =</span> <span class="fu">c</span>(<span class="st">'logi'</span>, <span class="st">'logi'</span>,<span class="st">'logi'</span>,<span class="st">'logi'</span>,<span class="st">'logi'</span>,<span class="st">'logi'</span>,</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>               <span class="st">'logi'</span>,<span class="st">'logi'</span>,<span class="st">'logi'</span>,<span class="st">'logi'</span>,<span class="st">'logi'</span>,<span class="st">'logi'</span>,<span class="st">'logi'</span>,<span class="st">'logi'</span>),                       </span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>   <span class="at">Variable =</span> <span class="fu">c</span>(<span class="st">'hair'</span>, <span class="st">'feathers'</span>,<span class="st">'eggs'</span>,<span class="st">'milk'</span>,<span class="st">'airborne'</span>,<span class="st">'aquatic'</span>,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>               <span class="st">'predator'</span>,<span class="st">'toothed'</span>,<span class="st">'backbone'</span>,<span class="st">'breathes'</span>,<span class="st">'venomous'</span>,</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>               <span class="st">'fins'</span>,<span class="st">'tail'</span>,<span class="st">'has_legs'</span>),</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>   <span class="at">Variable_Description =</span> <span class="fu">c</span>(<span class="st">'Has hair?'</span>,<span class="st">'has feathers?'</span>,<span class="st">'Lays eggs?'</span>,</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>                           <span class="st">'Produces milk?'</span>,<span class="st">'Can fly?'</span>,<span class="st">'Can swim?'</span>,</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>                           <span class="st">'Is a predator?'</span>,<span class="st">'Has teeth?'</span>,<span class="st">'Has a backbone?'</span>,</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>                           <span class="st">'Animal breathes?'</span>,<span class="st">'Has venom?'</span>,<span class="st">'Has fins?'</span>,</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>                           <span class="st">'Has a tail?'</span>,<span class="st">'Has legs?'</span>))</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>tab_lca_var_copy <span class="ot">&lt;-</span> tab_lca_var</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tab_lca_var_copy) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"# of Variables"</span>, <span class="st">"Data Type"</span>, <span class="st">"Variables"</span>, <span class="st">"Variable Description"</span>)</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="co">#Printing a table of variables used in LCA modeling</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(tab_lca_var_copy, <span class="at">format =</span> <span class="st">"html"</span>, <span class="at">caption =</span> <span class="st">"Variables considered in LCA Analysis"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"striped"</span>, <span class="st">"hover"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">background =</span> <span class="st">"steelblue"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">column_spec</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>, <span class="at">background =</span> <span class="st">"lavender"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">row_spec</span>(<span class="dv">0</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">background =</span> <span class="st">"darkblue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<caption>Variables considered in LCA Analysis</caption>
 <thead>
  <tr>
   <th style="text-align:right;font-weight: bold;color: white !important;background-color: darkblue !important;"> # of Variables </th>
   <th style="text-align:left;font-weight: bold;color: white !important;background-color: darkblue !important;"> Data Type </th>
   <th style="text-align:left;font-weight: bold;color: white !important;background-color: darkblue !important;"> Variables </th>
   <th style="text-align:left;font-weight: bold;color: white !important;background-color: darkblue !important;"> Variable Description </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 1 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> hair </td>
   <td style="text-align:left;background-color: lavender !important;"> Has hair? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 2 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> feathers </td>
   <td style="text-align:left;background-color: lavender !important;"> has feathers? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 3 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> eggs </td>
   <td style="text-align:left;background-color: lavender !important;"> Lays eggs? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 4 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> milk </td>
   <td style="text-align:left;background-color: lavender !important;"> Produces milk? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 5 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> airborne </td>
   <td style="text-align:left;background-color: lavender !important;"> Can fly? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 6 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> aquatic </td>
   <td style="text-align:left;background-color: lavender !important;"> Can swim? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 7 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> predator </td>
   <td style="text-align:left;background-color: lavender !important;"> Is a predator? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 8 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> toothed </td>
   <td style="text-align:left;background-color: lavender !important;"> Has teeth? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 9 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> backbone </td>
   <td style="text-align:left;background-color: lavender !important;"> Has a backbone? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 10 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> breathes </td>
   <td style="text-align:left;background-color: lavender !important;"> Animal breathes? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 11 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> venomous </td>
   <td style="text-align:left;background-color: lavender !important;"> Has venom? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 12 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> fins </td>
   <td style="text-align:left;background-color: lavender !important;"> Has fins? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 13 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> tail </td>
   <td style="text-align:left;background-color: lavender !important;"> Has a tail? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 14 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> has_legs </td>
   <td style="text-align:left;background-color: lavender !important;"> Has legs? </td>
  </tr>
</tbody>
</table>

</div>
</div>
<p><strong>Table 3.2.2 Variables in Original Dataset, but not used in LCA models:</strong> <em>These will be used in our assessment of LCA Model Quality</em></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>tab_nonlca_var <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">var_num =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>, </span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>   <span class="at">Variable_data_type =</span> <span class="fu">c</span>(<span class="st">'logi'</span>, <span class="st">'logi'</span>,<span class="st">'logi'</span>,<span class="st">'logi'</span>),</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>   <span class="at">Variable =</span> <span class="fu">c</span>(<span class="st">'animal'</span>, <span class="st">'domestic'</span>,<span class="st">'catsize'</span>,<span class="st">'type'</span>),</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">Variable_Description =</span> <span class="fu">c</span>(<span class="st">'Animal Name'</span>,<span class="st">'Commonly domesticated?'</span>,</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>                            <span class="st">'About the size of a cat?'</span>,</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>                           <span class="st">'Animal Classification'</span>))</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>tab_nonlca_var_copy <span class="ot">&lt;-</span> tab_nonlca_var</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tab_nonlca_var_copy) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"# of Variables"</span>, <span class="st">"Data Type"</span>, <span class="st">"Variables"</span>, <span class="st">"Variable Description"</span>)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Printing a table of variables not-used in LCA modeling</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(tab_nonlca_var_copy, <span class="at">format =</span> <span class="st">"html"</span>, <span class="at">caption =</span> <span class="st">"Variables in Original Dataset, but not used in LCA models:"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"striped"</span>, <span class="st">"hover"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">background =</span> <span class="st">"steelblue"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">column_spec</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>, <span class="at">background =</span> <span class="st">"lavender"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">row_spec</span>(<span class="dv">0</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">background =</span> <span class="st">"darkblue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<caption>Variables in Original Dataset, but not used in LCA models:</caption>
 <thead>
  <tr>
   <th style="text-align:right;font-weight: bold;color: white !important;background-color: darkblue !important;"> # of Variables </th>
   <th style="text-align:left;font-weight: bold;color: white !important;background-color: darkblue !important;"> Data Type </th>
   <th style="text-align:left;font-weight: bold;color: white !important;background-color: darkblue !important;"> Variables </th>
   <th style="text-align:left;font-weight: bold;color: white !important;background-color: darkblue !important;"> Variable Description </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 1 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> animal </td>
   <td style="text-align:left;background-color: lavender !important;"> Animal Name </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 2 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> domestic </td>
   <td style="text-align:left;background-color: lavender !important;"> Commonly domesticated? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 3 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> catsize </td>
   <td style="text-align:left;background-color: lavender !important;"> About the size of a cat? </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 4 </td>
   <td style="text-align:left;background-color: lavender !important;"> logi </td>
   <td style="text-align:left;background-color: lavender !important;"> type </td>
   <td style="text-align:left;background-color: lavender !important;"> Animal Classification </td>
  </tr>
</tbody>
</table>

</div>
</div>
</section>
<section id="load-data" class="level3">
<h3 class="anchored" data-anchor-id="load-data">3.3 Load Data</h3>
<p>In the original raw dataset, within each categorical variable text, there is an apostrophe and a “b” before each word and at the end, there is an apostrophe. The data were cleaned into the file Zoo_clean with these extra characters removed leaving only “TRUE” &amp; “FALSE”.</p>
<p>Below is where we import the cleaned dataset without the apostrophes or the “b”s.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>Zoo_clean<span class="ot">&lt;-</span> (<span class="fu">read.csv</span>(<span class="st">"https://raw.githubusercontent.com/kamstandhaft/STA6257_Project_LatentClassAnalysis/main/Zoo_clean.csv"</span>))</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>new_zoo <span class="ot">&lt;-</span> Zoo_clean <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">animal=</span><span class="cn">NULL</span>, <span class="at">domestic=</span><span class="cn">NULL</span>, <span class="at">catsize=</span><span class="cn">NULL</span>, <span class="at">type=</span><span class="cn">NULL</span>)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="fu">view</span>(new_zoo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="data-preparation" class="level3">
<h3 class="anchored" data-anchor-id="data-preparation">3.4 Data Preparation</h3>
<p>As mentioned above, we will be removing the “type” variable from the dataset with the goal of trying to replicate this level of classification through the latent class detection. The identifier variable “animal” will also be removed. Finally, we only want objective binary variables in our dataset, so we will also be removing “catsize” and “domestic” from the dataset, and transforming legs (denoting the number of legs a creature has) to has_legs (a binary operator indicating whether any legs are present or not).</p>
<p>We will be using the poLCA function from the poLCA package library, which requires an input of positive integer values &gt;=1. As such, we transformed our dataset from logical operators TRUE/FALSE to binary integers 0/1. But as poLCA function cannot accept “0” as a value, we further transformed the data by incrementing all values by 1, wherein “FALSE” is represented by “1”, and “TRUE” is represented by “2”.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Creating "has_legs" to denote whether the animal has a leg count &gt;0 from "legs" variable</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>new_zoo <span class="ot">&lt;-</span> Zoo_clean <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">has_legs =</span> <span class="fu">ifelse</span>(legs <span class="sc">==</span> <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>))</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Removing "legs" since it is continuous and not binary</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>new_zoo <span class="ot">&lt;-</span> new_zoo <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">legs =</span> <span class="cn">NULL</span>)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Converting has_legs from integer to logical to match the rest of the dataset</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>new_zoo<span class="sc">$</span>has_legs <span class="ot">&lt;-</span> <span class="fu">as.logical</span>(new_zoo<span class="sc">$</span>has_legs)</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Removing "animal","domestic", "catsize", and "type" from 18-variable dataset to make new 14-variable dataset for LCA modeling  </span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>new_zoo_subset <span class="ot">&lt;-</span> new_zoo <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">animal=</span><span class="cn">NULL</span>, <span class="at">domestic=</span><span class="cn">NULL</span>, </span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">catsize=</span><span class="cn">NULL</span>, <span class="at">type=</span><span class="cn">NULL</span>)</span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="co">#Creating a dataset where the binary data is converted to integers 0/"FALSE" or 1/"TRUE"</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>new_zoo_int <span class="ot">&lt;-</span> new_zoo_subset <span class="sc">%&gt;%</span> <span class="fu">mutate_all</span>(<span class="sc">~</span><span class="fu">as.integer</span>(.))</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="co">#Incrementing the binary integers from 0/1 to 1/2 - poLCA() can only accept integers &gt;=1</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>new_zoo_int_1 <span class="ot">&lt;-</span> new_zoo_int <span class="sc">%&gt;%</span> <span class="fu">mutate_all</span>(<span class="sc">~</span>. <span class="sc">+</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="statistical-modeling" class="level2">
<h2 class="anchored" data-anchor-id="statistical-modeling">4. Statistical Modeling</h2>
<p>Potentially the most difficult decision to make regarding identification of latent classes is determining the correct number of classes to use in our model. We will be assessing 2,3,4,5,6, and 7 number of classes in modeling.</p>
<p>Below we begin preparing to build our models by using the cbind function used to bind Columns being used in LCA.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># cbind function used to bind Columns being used in LCA</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>lca_bind <span class="ot">&lt;-</span>  <span class="fu">cbind</span>(hair, feathers, eggs, milk, </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>                   airborne, aquatic, predator, </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>                   toothed, backbone, breathes, </span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>                   venomous, fins, tail, has_legs) <span class="sc">~</span> <span class="dv">1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="fitting-lca-models-with-2-7-of-classes" class="level3">
<h3 class="anchored" data-anchor-id="fitting-lca-models-with-2-7-of-classes">4.1 Fitting LCA Models with 2-7 # of Classes</h3>
<p>We will be running each of our 6 models 100 times(100 random starting values) each with 100 iterations. This will allow us to have consistent results for each number of classes per model to report upon without large swings in class sizes and predictions, and will additionally provide a best model for each number of classes.</p>
<p><strong>Fitting a 2-class LCA Model</strong></p>
<p>For the 2-class model, we have selected to display the output of the poLCA model results using the [graph = TRUE] option.</p>
<p>For each variable, this indicates the percentage of members of the original dataset that have that attribute broken down by which latent class was assigned in the model for that number of classes.</p>
<p>We chose to assess up to 7-class models because the original # of classifications was 7 “types”.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>lca_fit2 <span class="ot">&lt;-</span> <span class="fu">poLCA</span>(lca_bind, <span class="at">data =</span> new_zoo_int_1, </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">nclass =</span> <span class="dv">2</span>, <span class="at">graphs =</span> <span class="cn">FALSE</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>, </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">nrep=</span><span class="dv">100</span>, <span class="at">maxiter=</span><span class="dv">100</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/2_class_model.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Figure 4.1.1 - LCA 2-Class Model</figcaption><p></p>
</figure>
</div>
<p>(Figure 4.1.1) shows the graphical output of the poLCA model when 2-Classes are specified. The proportions at the base of the plot (0.3953 &amp; 0.6047) indicate the percentage of the original dataset that the poLCA function has classified into each of the detected latent classes.</p>
<p>The vertical red bars [Pr(outcome)] denote the percentage of the members of that assigned class have that particular attribute. While the proportions at the base of the plot should always add to 1.0, the member percentage of each assigned class may have any value from 0-1.0 across all classes for that attribute. The poLCA function will additionally output a proportion value between 0 and 1 denoting the likelihood that each attribute will be included within each class generated by the function, as seen in (Figure 4.1.2) below.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/2_class_prop_output.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Figure 4.1.2 - LCA 2-Class Proportion Output</figcaption><p></p>
</figure>
</div>
<p><strong>Fitting a 3-class LCA Model</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>lca_fit3 <span class="ot">&lt;-</span> <span class="fu">poLCA</span>(lca_bind, <span class="at">data =</span> new_zoo_int_1, </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">nclass =</span> <span class="dv">3</span>, <span class="at">graphs =</span> <span class="cn">FALSE</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>, </span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">verbose =</span> <span class="cn">FALSE</span>, <span class="at">nrep=</span><span class="dv">100</span>, <span class="at">maxiter=</span><span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>Fitting a 4-class LCA Model</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>lca_fit4 <span class="ot">&lt;-</span> <span class="fu">poLCA</span>(lca_bind, <span class="at">data =</span> new_zoo_int_1, </span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">nclass =</span> <span class="dv">4</span>, <span class="at">graphs =</span> <span class="cn">FALSE</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>, </span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">verbose =</span> <span class="cn">FALSE</span>, <span class="at">nrep=</span><span class="dv">100</span>, <span class="at">maxiter=</span><span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>Fitting a 5-class LCA Model</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>lca_fit5 <span class="ot">&lt;-</span> <span class="fu">poLCA</span>(lca_bind, <span class="at">data =</span> new_zoo_int_1, </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">nclass =</span> <span class="dv">5</span>, <span class="at">graphs =</span> <span class="cn">FALSE</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">verbose =</span> <span class="cn">FALSE</span>, <span class="at">nrep=</span><span class="dv">100</span>, <span class="at">maxiter=</span><span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>Fitting a 6-class LCA Model</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>lca_fit6 <span class="ot">&lt;-</span> <span class="fu">poLCA</span>(lca_bind, <span class="at">data =</span> new_zoo_int_1, </span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">nclass =</span> <span class="dv">6</span>, <span class="at">graphs =</span> <span class="cn">FALSE</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">verbose =</span> <span class="cn">FALSE</span>, <span class="at">nrep=</span><span class="dv">100</span>, <span class="at">maxiter=</span><span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>Fitting a 7-class LCA Model</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>lca_fit7 <span class="ot">&lt;-</span> <span class="fu">poLCA</span>(lca_bind, <span class="at">data =</span> new_zoo_int_1, </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">nclass =</span> <span class="dv">7</span>, <span class="at">graphs =</span> <span class="cn">FALSE</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">verbose =</span> <span class="cn">FALSE</span>, <span class="at">nrep=</span><span class="dv">100</span>, <span class="at">maxiter=</span><span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="model-comparison-statistics" class="level3">
<h3 class="anchored" data-anchor-id="model-comparison-statistics">4.2 Model Comparison Statistics</h3>
<p>As determining the correct number of classes to use to properly model the data can often prove challenging, we will use a number of statistics to help indicate fit quality for each model, which we go through below to help select the correct number of classes for a good model.</p>
<p><strong>Note</strong>: For each of these 6 models, we are running 100 repetitions random starting value, each with 100 iterations to help achieve convergence of the models for comparison. The criteria for which the best model is chosen by the poLCA function is the lowest BIC value. Without increasing the number of runs and iterations, the model and fit statistics can be inconsistent.</p>
<p>To compare these models, we will be primarily looking at the Akaike Information Criterion (AIC), the Bayesian Information Criterion (BIC), and the Chi-square goodness of fit statistic (X^2).</p>
<p>In a traditional research environment, these statistics would be likely the best decision criteria we could expect to have for assessing the model. In the case of our data, we have a comparison classification attribute “type” in our dataset that we will withhold from the Latent Class Analysis procedure as a secondary means to assess how well the model classifies the members of the dataset - in this case animals within a zoo. It is possible, however, that the LCA may find latent classes that differ from these defined classifications, which roughly correspond to the Linnaean taxonomic classification of classes within the Kingdom Animalia.</p>
<p>A lower value for the AIC and BIC values as well as the (X^2), indicate that the model is a better fit for the data.</p>
<p>As illustrated in the figure below,the 5-class, 6-class, and 7-class models all have substantially lower values for these statistical metrics than for the 2-class, 3-class, and 4-class models. It is worth noting however, that the 7-class model posts an alert indicating that the number of parameters estimated exceeds the number of observations, which effectively provides a negative residual Degree of Freedom (DoF). For this reason, we can eliminate the 7-class model.</p>
<p><strong>Table 4.2.1 Comparison of AIC, BIC, and Goodness of Fit for each # of Classes</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load Data</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="co">#![Figure 4.2.1 Model Comparison](index_files/figure-html/Model_comparison.PNG)</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>class_count <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">6</span>, <span class="dv">7</span>)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>fit_stats_long <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">class_count =</span> <span class="fu">rep</span>(class_count, <span class="dv">2</span>),</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">fit_stat =</span>       <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"AIC"</span>, <span class="fu">length</span>(class_count)), <span class="fu">rep</span>(<span class="st">"BIC"</span>, <span class="fu">length</span>(class_count))),</span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">fit_stat_value =</span> <span class="fu">c</span>(lca_fit2<span class="sc">$</span>aic, lca_fit3<span class="sc">$</span>aic, lca_fit4<span class="sc">$</span>aic, lca_fit5<span class="sc">$</span>aic, lca_fit6<span class="sc">$</span>aic, lca_fit7<span class="sc">$</span>aic,</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>                   lca_fit2<span class="sc">$</span>bic, lca_fit3<span class="sc">$</span>bic, lca_fit4<span class="sc">$</span>bic, lca_fit5<span class="sc">$</span>bic, lca_fit6<span class="sc">$</span>bic, lca_fit7<span class="sc">$</span>bic))</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>fit_AIC <span class="ot">&lt;-</span> <span class="fu">c</span>(lca_fit2<span class="sc">$</span>aic, lca_fit3<span class="sc">$</span>aic, lca_fit4<span class="sc">$</span>aic, lca_fit5<span class="sc">$</span>aic, lca_fit6<span class="sc">$</span>aic, lca_fit7<span class="sc">$</span>aic)</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>fit_BIC <span class="ot">&lt;-</span> <span class="fu">c</span>(lca_fit2<span class="sc">$</span>bic, lca_fit3<span class="sc">$</span>bic, lca_fit4<span class="sc">$</span>bic, lca_fit5<span class="sc">$</span>bic, lca_fit6<span class="sc">$</span>bic, lca_fit7<span class="sc">$</span>bic)</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>fit_GoF <span class="ot">&lt;-</span> <span class="fu">c</span>(lca_fit2<span class="sc">$</span>Chisq, lca_fit3<span class="sc">$</span>Chisq, lca_fit4<span class="sc">$</span>Chisq, lca_fit5<span class="sc">$</span>Chisq, lca_fit6<span class="sc">$</span>Chisq, lca_fit7<span class="sc">$</span>Chisq)</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>fit_stats <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(class_count, fit_AIC, fit_BIC, fit_GoF)</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a><span class="co">#Making a copy with easier to read column names</span></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>fit_stats_copy <span class="ot">&lt;-</span> fit_stats</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(fit_stats_copy) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Class Count"</span>, <span class="st">"AIC"</span>, <span class="st">"BIC"</span>, <span class="st">"Goodness of Fit"</span>)</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Using the knitr library</span></span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(fit_stats_copy, <span class="at">format =</span> <span class="st">"html"</span>, <span class="at">caption =</span> <span class="st">"Fit Statistics Table"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"striped"</span>, <span class="st">"hover"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">background =</span> <span class="st">"steelblue"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>, <span class="at">background =</span> <span class="st">"lavender"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row_spec</span>(<span class="dv">0</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">background =</span> <span class="st">"darkblue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<caption>Fit Statistics Table</caption>
 <thead>
  <tr>
   <th style="text-align:right;font-weight: bold;color: white !important;background-color: darkblue !important;"> Class Count </th>
   <th style="text-align:right;font-weight: bold;color: white !important;background-color: darkblue !important;"> AIC </th>
   <th style="text-align:right;font-weight: bold;color: white !important;background-color: darkblue !important;"> BIC </th>
   <th style="text-align:right;font-weight: bold;color: white !important;background-color: darkblue !important;"> Goodness of Fit </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 2 </td>
   <td style="text-align:right;background-color: lavender !important;"> 1275.0796 </td>
   <td style="text-align:right;background-color: lavender !important;"> 1350.918 </td>
   <td style="text-align:right;background-color: lavender !important;"> 21906.549 </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 3 </td>
   <td style="text-align:right;background-color: lavender !important;"> 1102.0079 </td>
   <td style="text-align:right;background-color: lavender !important;"> 1217.073 </td>
   <td style="text-align:right;background-color: lavender !important;"> 8660.619 </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 4 </td>
   <td style="text-align:right;background-color: lavender !important;"> 1010.0118 </td>
   <td style="text-align:right;background-color: lavender !important;"> 1164.304 </td>
   <td style="text-align:right;background-color: lavender !important;"> 5109.766 </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 5 </td>
   <td style="text-align:right;background-color: lavender !important;"> 983.2301 </td>
   <td style="text-align:right;background-color: lavender !important;"> 1176.749 </td>
   <td style="text-align:right;background-color: lavender !important;"> 2488.471 </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 6 </td>
   <td style="text-align:right;background-color: lavender !important;"> 965.9335 </td>
   <td style="text-align:right;background-color: lavender !important;"> 1198.679 </td>
   <td style="text-align:right;background-color: lavender !important;"> 2283.357 </td>
  </tr>
  <tr>
   <td style="text-align:right;font-weight: bold;color: white !important;background-color: steelblue !important;"> 7 </td>
   <td style="text-align:right;background-color: lavender !important;"> 958.7507 </td>
   <td style="text-align:right;background-color: lavender !important;"> 1230.723 </td>
   <td style="text-align:right;background-color: lavender !important;"> 1224.750 </td>
  </tr>
</tbody>
</table>

</div>
</div>
<p>The 5-class model has a better fit according to the BIC(1177) values compared to the 6-class model BIC(1199), but the AIC for the 6-class model (966) is slightly better than the 5-class model AIC(983).</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> fit_stats_long, <span class="fu">aes</span>(<span class="at">x =</span> class_count, <span class="at">y =</span> fit_stat_value, <span class="at">color =</span> fit_stat)) <span class="sc">+</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color=</span><span class="st">'black'</span>) <span class="sc">+</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"AIC and BIC Comparison"</span>,</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Number of Classes"</span>,</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Fit Statistics Value"</span>,</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"Fit Statistics"</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_discrete</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"AIC"</span>, <span class="st">"BIC"</span>)) <span class="sc">+</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># + facet_wrap(~ fit_stat, ncol = 1)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>Figure 4.2.1 - Comparison of AIC and BIC across 2, 3, 4, 5, 6, and 7-Class Models</strong></p>
<p>We additionally look at the Chi-square goodness of fit statistic (X^2). The GoF(X^2) for the 5-class model(1858) is better however than the (X^2) for the 6-class model(2283).</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Goodness of Fit</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>fit_GoF_stat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(class_count, fit_GoF)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> fit_GoF_stat, <span class="fu">aes</span>(<span class="at">x =</span> class_count, <span class="at">y =</span> fit_GoF)) <span class="sc">+</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">'green'</span>) <span class="sc">+</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Goodness of Fit"</span>,</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Number of Classes"</span>,</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Goodness of Fit Statistic Value"</span>) <span class="sc">+</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<p><img src="index_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><strong>Figure 4.2.2 - Comparison of GoF Values across 2, 3, 4, 5, 6, and 7-Class Models</strong></p>
<p>Without a clearly better model between these two, we can further compare their entropy (see below).</p>
<p>The entropy of a model speaks to how well the classes are separated with a higher entropy value indicating a likely increase in the separation of classes.</p>
<p>As we see below, the 5-class model has a slightly higher entropy value than the 6-class model implying a slightly higher separability of classes. With BIC, GoF, and entropy in favor of the 5-class model, we select this model to use to assess and illustrate the classification schema that LCA uses to classify the dataset.</p>
<p><strong>Table 4.2.3 Comparison of Entropy Values for the 5-Class and 6-Class Models</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>lca_fit5.ent <span class="ot">&lt;-</span> <span class="fu">poLCA.entropy</span>(lca_fit5)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>lca_fit6.ent <span class="ot">&lt;-</span> <span class="fu">poLCA.entropy</span>(lca_fit6)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>tab_lca_entropy <span class="ot">&lt;-</span> <span class="fu">data.frame</span>( </span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>   <span class="at">Model_Class_Number =</span> <span class="fu">c</span>(<span class="st">'5-Class Model'</span>, <span class="st">'6-Class Model'</span>),</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>   <span class="at">Entropy =</span> <span class="fu">c</span>(lca_fit5.ent,lca_fit6.ent))</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>tab_lca_entropy_copy <span class="ot">&lt;-</span> tab_lca_entropy</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tab_lca_entropy_copy) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Model Classes"</span>, <span class="st">"Entropy Value"</span>)</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Using the knitr library</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(tab_lca_entropy_copy, <span class="at">format =</span> <span class="st">"html"</span>, <span class="at">caption =</span> <span class="st">"Entropy Comparison of 5-Class and 6-Class Models"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"striped"</span>, <span class="st">"hover"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">background =</span> <span class="st">"steelblue"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">2</span>, <span class="at">background =</span> <span class="st">"lavender"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row_spec</span>(<span class="dv">0</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">background =</span> <span class="st">"darkblue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<caption>Entropy Comparison of 5-Class and 6-Class Models</caption>
 <thead>
  <tr>
   <th style="text-align:left;font-weight: bold;color: white !important;background-color: darkblue !important;"> Model Classes </th>
   <th style="text-align:right;font-weight: bold;color: white !important;background-color: darkblue !important;"> Entropy Value </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;font-weight: bold;color: white !important;background-color: steelblue !important;"> 5-Class Model </td>
   <td style="text-align:right;background-color: lavender !important;"> 4.136046 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;color: white !important;background-color: steelblue !important;"> 6-Class Model </td>
   <td style="text-align:right;background-color: lavender !important;"> 3.902091 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</section>
<section id="chosen-model-discussion-and-assessment" class="level3">
<h3 class="anchored" data-anchor-id="chosen-model-discussion-and-assessment">4.3 Chosen Model Discussion and Assessment</h3>
<p>With the better AIC, GoF(X^2), and Entropy value, we selected the 5-Class model as our preferred model.</p>
<p>The 5-class population proportion graph shown below (Figure 4.3.1) displays along the x-axis, the proportions of the total dataset population that were classified into each of the 5-class of these model.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/5_class_ordered.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Figure 4.3.1 - 5-Class Model</figcaption><p></p>
</figure>
</div>
<p>Under a normal research environment, we would not likely have the true classifications to compare these classes to against their predicted counterparts. But as we chose a dataset for which we could remove classification values and have the LCA models assess the attributes of the members, we designed the experiment to illustrate how Latent Class Analysis predicts which class a member should belong to, and can compare these to against a real classification schema.</p>
<p>For the purposes of these figures, we filtered selections based on whether the attributes had above ~70% likelihood of being members in the class according to the model. By doing so, there is the potential for overlap with certain member showing in more than one class, but with that in mind, this approach has the benefit of showing how potentially effective LCA can be at detecting these latent groups.</p>
<p><strong>Table 4.3.1 Comparison of AIC, BIC, and Goodness of Fit for each # of Classes</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Reordering the Model output graph to display in order from highest to lowest proportion for easier comparison and labeling</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>probs.start.new <span class="ot">&lt;-</span> <span class="fu">poLCA.reorder</span>(lca_fit5<span class="sc">$</span>probs.start,<span class="fu">order</span>(lca_fit5<span class="sc">$</span>P,<span class="at">decreasing=</span><span class="cn">TRUE</span>))</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>lca_fit5 <span class="ot">&lt;-</span> <span class="fu">poLCA</span>(lca_bind, <span class="at">data =</span> new_zoo_int_1,<span class="at">nclass=</span><span class="dv">5</span>,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">graphs =</span> <span class="cn">FALSE</span>,<span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">verbose =</span> <span class="cn">FALSE</span>, <span class="at">nrep=</span><span class="dv">100</span>, <span class="at">maxiter=</span><span class="dv">100</span>,</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">probs.start=</span>probs.start.new)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>orig_classes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>( </span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>   <span class="at">Class =</span> <span class="fu">c</span>(<span class="st">'Class 1'</span>, <span class="st">'Class 2'</span>, <span class="st">'Class 3'</span>, <span class="st">'Class 4'</span>, <span class="st">'Class 5'</span>),</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>   <span class="at">Percentage =</span> <span class="fu">c</span>(<span class="st">'35.46%'</span>, <span class="st">'20.78%'</span>,<span class="st">'17.82%'</span>,<span class="st">'13.09%'</span>,<span class="st">'12.85%'</span>))</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(orig_classes, <span class="at">format =</span> <span class="st">"html"</span>, <span class="at">caption =</span> <span class="st">"Population Percentages of each Class in the 5-Class Model"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"striped"</span>, <span class="st">"hover"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">background =</span> <span class="st">"steelblue"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">2</span>, <span class="at">background =</span> <span class="st">"lavender"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row_spec</span>(<span class="dv">0</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">background =</span> <span class="st">"darkblue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<caption>Population Percentages of each Class in the 5-Class Model</caption>
 <thead>
  <tr>
   <th style="text-align:left;font-weight: bold;color: white !important;background-color: darkblue !important;"> Class </th>
   <th style="text-align:left;font-weight: bold;color: white !important;background-color: darkblue !important;"> Percentage </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;font-weight: bold;color: white !important;background-color: steelblue !important;"> Class 1 </td>
   <td style="text-align:left;background-color: lavender !important;"> 35.46% </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;color: white !important;background-color: steelblue !important;"> Class 2 </td>
   <td style="text-align:left;background-color: lavender !important;"> 20.78% </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;color: white !important;background-color: steelblue !important;"> Class 3 </td>
   <td style="text-align:left;background-color: lavender !important;"> 17.82% </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;color: white !important;background-color: steelblue !important;"> Class 4 </td>
   <td style="text-align:left;background-color: lavender !important;"> 13.09% </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;color: white !important;background-color: steelblue !important;"> Class 5 </td>
   <td style="text-align:left;background-color: lavender !important;"> 12.85% </td>
  </tr>
</tbody>
</table>

</div>
</div>
<p><strong>Class 1</strong> which, as the largest class, is estimated to contain ~35.46% of the population predicts its members to have ‘hair’, ‘toothed’, ‘backbone’, ‘breathes’, ‘tail’, ‘has_legs’. We filtered the cleaned new_zoo dataset for these attributes, and noted that all members were classified as type ‘mammal’ in the original dataset.</p>
<p><strong>Class 2</strong> which is estimated to contain ~20.78% of the population predicts its members to have ‘feathers’, ‘eggs’, ‘backbone’, ‘breathes’, ‘tail’, ‘has_legs’. We filtered the cleaned new_zoo dataset for these attributes, and noted that all members were classified as type ‘bird’ in the original dataset.</p>
<p><strong>Class 3</strong> is estimated to contain ~17.82% of the population. It should be noted that it has two attributes (‘breathes’ and ‘predator’) which were both over 50% shown on the graph, but as we excluded our initial filtering assessment to ~80% and above, these are not selected in the filter.</p>
<p>These class members are predicted to have ‘eggs’ and ‘has_legs’. While this limited selection duplicates a few of the previous members that were selected for other classes, this class also contains the invertebrates and insects, which were not present in the other classes. Based solely upon outside comparison to the type attribute in the original dataset, this class appears to be the least homogenous.</p>
<p><strong>Class 4</strong> which is estimated to contain ~13.09% of the population is less homogenous than Classes 1 &amp; 2, but not as heterogenous as Class 3. The original classification schema of “type” had 7 unique values, which implies that with a 5-class model, overlap is inevitable. This is only visible due to the fact that we have the original data for “type” for illustration purposes, which is unlikely to be the case in research data.This class’ members ar predicted to have ‘aquatic’, ‘predator’, ‘toothed’, ‘backbone’, ‘breathes’, ‘tail’. We filtered the cleaned new_zoo dataset for these attributes. The resulting filter only yielded 5 members, which are mostly aquatic mammals and a single amphibian.</p>
<p><strong>Class 5</strong> which is the smallest class, estimated to contain ~12.85% of the population predicts its members to have ‘eggs’, ‘aquatic’, ‘toothed’, ‘backbone’, ‘fins’, ‘tail’. We filtered the cleaned new_zoo dataset for these attributes, and noted that all members were classified as type ‘fish’ in the original dataset.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Here we are converting the new_zoo_int_2 into a factor so it can be used with MCA() function</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>new_zoo_int_1_factor <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(new_zoo_int_1, factor))</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Saving the result of MCA function applied to new_zoo_int_2_factor into mca_outcome</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Made invisible so the output does not print instead of image with label</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>mca_outcome <span class="ot">&lt;-</span> <span class="fu">invisible</span>(<span class="fu">MCA</span>(new_zoo_int_1_factor, <span class="at">graph =</span> <span class="cn">FALSE</span>))</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Setting the # of dimension in the biplot to 2 (just x and y axis)</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>n_dimensions <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Creating the MCA biplot from mca_outcome</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>mca_biplot <span class="ot">&lt;-</span> <span class="fu">fviz_mca_biplot</span>(</span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>  mca_outcome,</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">repel =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">axes =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="co"># we are only displaying axes 1(Dim1)&amp; 2(Dim2)</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">title =</span> <span class="fu">paste</span>(<span class="st">"MCA Biplot (Dimensions:"</span>, n_dimensions, <span class="st">")"</span>))</span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a><span class="co">#Printing the MCA biplot</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a><span class="co">#mca_biplot</span></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a><span class="co">#----</span></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Extract the class membership probabilities and convert to dataframe</span></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a>class_probability <span class="ot">&lt;-</span> lca_fit5<span class="sc">$</span>posterior</span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a>class_probability_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(class_probability)</span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a><span class="co">#Build a matrix from the dataframe and assign each observation to the LCA class with the highest probability</span></span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a>class_probability_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(class_probability_df)</span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a>lca_result_classes <span class="ot">&lt;-</span> <span class="fu">max.col</span>(class_probability_matrix, <span class="at">ties.method =</span> <span class="st">"random"</span>)</span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a><span class="co">#Turn lca_result_classes into a factor for MCA input</span></span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a>lca_result_class_factor <span class="ot">&lt;-</span> <span class="fu">factor</span>(lca_result_classes, <span class="at">levels =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Class 1"</span>, <span class="st">"Class 2"</span>, <span class="st">"Class 3"</span>, <span class="st">"Class 4"</span>, <span class="st">"Class 5"</span>))</span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a><span class="co">#Find the coordinate points generated by the Multiple Correspondence Analysis (MCA) function</span></span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a>coord_pts <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(mca_outcome<span class="sc">$</span>ind<span class="sc">$</span>coord)</span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a><span class="co">#Adding the LCA class labels to the dataframe with the coordinates</span></span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a>coord_pts<span class="sc">$</span>lca_result_class_factor <span class="ot">&lt;-</span> lca_result_class_factor</span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a><span class="co">#Plot the LCA classes as ellipses over the MCA plot of plots and save as LCA_MCA_biplot</span></span>
<span id="cb19-41"><a href="#cb19-41" aria-hidden="true" tabindex="-1"></a>LCA_MCA_biplot <span class="ot">&lt;-</span></span>
<span id="cb19-42"><a href="#cb19-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(coord_pts, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Dim 1</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Dim 2</span><span class="st">`</span>, <span class="at">color =</span> lca_result_class_factor)) <span class="sc">+</span></span>
<span id="cb19-43"><a href="#cb19-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb19-44"><a href="#cb19-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_ellipse</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">Dim 1</span><span class="st">`</span>, <span class="at">y =</span> <span class="st">`</span><span class="at">Dim 2</span><span class="st">`</span>, <span class="at">color =</span> lca_result_class_factor), <span class="at">type =</span> <span class="st">"norm"</span>, <span class="at">level =</span> <span class="fl">0.95</span>) <span class="sc">+</span></span>
<span id="cb19-45"><a href="#cb19-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">rownames</span>(coord_pts)), <span class="at">nudge_x =</span> <span class="fl">0.02</span>, <span class="at">nudge_y =</span> <span class="fl">0.02</span>, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">check_overlap =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb19-46"><a href="#cb19-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb19-47"><a href="#cb19-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"MCA Biplot with LCA Classes"</span>,</span>
<span id="cb19-48"><a href="#cb19-48" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Dimension 1 (31.2% Variance Explained by this Dimension)"</span>,</span>
<span id="cb19-49"><a href="#cb19-49" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Dimension 2 (25.3% Variance Explained by this Dimension)"</span>,</span>
<span id="cb19-50"><a href="#cb19-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"LCA Class"</span>)</span>
<span id="cb19-51"><a href="#cb19-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-52"><a href="#cb19-52" aria-hidden="true" tabindex="-1"></a><span class="co">#LCA_MCA_biplot</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/mca_biplot_lca.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Figure 4.3.2 - MCA Biplot with 5 Class LCA Model Classes Overlain</figcaption><p></p>
</figure>
</div>
</section>
<section id="sensitivity-analysis---correlation" class="level3">
<h3 class="anchored" data-anchor-id="sensitivity-analysis---correlation">4.4 Sensitivity Analysis - Correlation</h3>
<p>Researchers have the ability to modify the model specification by either adding or removing variables or constraints to assess the sensitivity of the results to these changes. One of the major assumptions when performing Latent Class Analysis (LCA) is is an independence of the variables within a given class detected; variable redundancy and multicollinearity therefore becomes an important factor to consider, particularly when correlation coefficients for specific variables are greater than 0.50 <span class="citation" data-cites="sinha2021practitioner">(<a href="#ref-sinha2021practitioner" role="doc-biblioref">Sinha, Calfee, and Delucchi 2021</a>)</span>. To analyse and assess the extent of this risk, we generated a correlation matrix and heatmap to assess the presence of multicollinearity, with a focus particularly on positively correlated variables that are present within the same classes, since by virtue of the underlying function of LCA, negatively correlated variables are less likely to appear within the same class.</p>
<p>We then test the model by removing these variables and comparing the new model to the previous model which includes these variables. We will evaluate whether the new model is still able to fit the data well and compare its goodness of fit statistics (such as AIC, BIC) to those of the previous model. If the new model shows improvement in goodness of fit, we may consider selecting it as our final model.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/Heatmap.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Figure 4.4.1 - Correlation heatmap for sensitivity analysis</figcaption><p></p>
</figure>
</div>
<p>As seen in the correlation heatmap (Figure 4.4.1), milk is highly correlated with both hair(0.88)and egg(-0.94). We decided to remove milk as an attribute, and re-consider our LCA model to see if it improves without this highly collinear variables included.</p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Creating a dataset without 'milk' - 13-variables</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>new_zoo_int_2 <span class="ot">&lt;-</span> new_zoo_int_1  <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">milk=</span><span class="cn">NULL</span>)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Binding 13-variables into columns for modified LCA model without 'milk'</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>lca_bind_corr <span class="ot">&lt;-</span>  <span class="fu">cbind</span>(hair, feathers, eggs, </span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>                   airborne, aquatic, predator, </span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>                   toothed, backbone, breathes, </span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>                   venomous, fins, tail, has_legs) <span class="sc">~</span> <span class="dv">1</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>lca_fit5_corr <span class="ot">&lt;-</span> <span class="fu">poLCA</span>(lca_bind_corr, <span class="at">data =</span> new_zoo_int_2, </span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">nclass =</span> <span class="dv">5</span>, <span class="at">graphs =</span> <span class="cn">FALSE</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">verbose =</span> <span class="cn">FALSE</span>, <span class="at">nrep=</span><span class="dv">100</span>, <span class="at">maxiter=</span><span class="dv">100</span>)</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a><span class="co">#Reordering the Model output graph to display in order from highest to lowest proprtion for easier comparison and labeling</span></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>probs.start.new2 <span class="ot">&lt;-</span> <span class="fu">poLCA.reorder</span>(lca_fit5_corr<span class="sc">$</span>probs.start,<span class="fu">order</span>(lca_fit5_corr<span class="sc">$</span>P,<span class="at">decreasing=</span><span class="cn">TRUE</span>))</span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>lca_fit5_corr <span class="ot">&lt;-</span> <span class="fu">poLCA</span>(lca_bind_corr, <span class="at">data =</span> new_zoo_int_2,<span class="at">nclass=</span><span class="dv">5</span>,</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>                  <span class="at">graphs =</span> <span class="cn">FALSE</span>,<span class="at">na.rm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>                  <span class="at">verbose =</span> <span class="cn">FALSE</span>, <span class="at">nrep=</span><span class="dv">100</span>,</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>                  <span class="at">probs.start=</span>probs.start.new2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="index_files/figure-html/lca_fit5_corr_graph.png" class="img-fluid figure-img"></p>
<p></p><figcaption class="figure-caption">Figure 4.4.2 - 5-Class LCA Model Population Percentages without ‘milk’</figcaption><p></p>
</figure>
</div>
<p>With the ‘milk’ variable removed from our LCA models due to high correlation values with 2 other attributes, our 5-Class models exhibited substantial improvement in AIC, BIC, and GoF, and slightly worse values for entropy. As a result, we find the adjusted 5-class model, with 13 attributes assessed to be our best final model.</p>
<p><strong>Table 4.4.1 Comparison Statistics for Original and Modified 5-Class Models</strong></p>
<div class="cell">
<details>
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>lca_fit5_corr.ent <span class="ot">&lt;-</span> <span class="fu">poLCA.entropy</span>(lca_fit5_corr)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Creating data frame of AIC, BIC, GoF, and Entropy of both original and modified 5-Class Models</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>class5_compare <span class="ot">&lt;-</span> <span class="fu">data.frame</span>( </span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>   <span class="at">Model =</span> <span class="fu">c</span>(<span class="st">'Original 5-Class Model'</span>, <span class="st">'Adjusted 5-Class Model'</span>),</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>   <span class="at">AIC =</span> <span class="fu">c</span>(lca_fit5<span class="sc">$</span>aic, lca_fit5_corr<span class="sc">$</span>aic),</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>   <span class="at">BIC =</span> <span class="fu">c</span>(lca_fit5<span class="sc">$</span>bic, lca_fit5_corr<span class="sc">$</span>bic),</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>   <span class="at">GoF =</span> <span class="fu">c</span>(lca_fit5<span class="sc">$</span>Chisq, lca_fit5_corr<span class="sc">$</span>Chisq),</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>   <span class="at">entropy =</span> <span class="fu">c</span>(lca_fit5.ent, lca_fit5_corr.ent))</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Printing table comparing original and modified 5-class stats</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a><span class="fu">kable</span>(class5_compare, <span class="at">format =</span> <span class="st">"html"</span>, <span class="at">caption =</span> <span class="st">"Comparison Statistics for Original and Modified 5-Class Models"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"striped"</span>, <span class="st">"hover"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">background =</span> <span class="st">"steelblue"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">2</span>, <span class="at">background =</span> <span class="st">"lavender"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">row_spec</span>(<span class="dv">0</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="at">background =</span> <span class="st">"darkblue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">

<table class="table table-striped table-hover" style="margin-left: auto; margin-right: auto;">
<caption>Comparison Statistics for Original and Modified 5-Class Models</caption>
 <thead>
  <tr>
   <th style="text-align:left;font-weight: bold;color: white !important;background-color: darkblue !important;"> Model </th>
   <th style="text-align:right;font-weight: bold;color: white !important;background-color: darkblue !important;"> AIC </th>
   <th style="text-align:right;font-weight: bold;color: white !important;background-color: darkblue !important;"> BIC </th>
   <th style="text-align:right;font-weight: bold;color: white !important;background-color: darkblue !important;"> GoF </th>
   <th style="text-align:right;font-weight: bold;color: white !important;background-color: darkblue !important;"> entropy </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;font-weight: bold;color: white !important;background-color: steelblue !important;"> Original 5-Class Model </td>
   <td style="text-align:right;background-color: lavender !important;"> 983.2301 </td>
   <td style="text-align:right;"> 1176.749 </td>
   <td style="text-align:right;"> 2488.471 </td>
   <td style="text-align:right;"> 4.136046 </td>
  </tr>
  <tr>
   <td style="text-align:left;font-weight: bold;color: white !important;background-color: steelblue !important;"> Adjusted 5-Class Model </td>
   <td style="text-align:right;background-color: lavender !important;"> 954.2627 </td>
   <td style="text-align:right;"> 1134.706 </td>
   <td style="text-align:right;"> 1568.740 </td>
   <td style="text-align:right;"> 4.043642 </td>
  </tr>
</tbody>
</table>

</div>
</div>
</section>
<section id="conclusion" class="level3">
<h3 class="anchored" data-anchor-id="conclusion">Conclusion</h3>
<p>Our research indicates that Latent Class Analysis (LCA) is an exceptionally effective tool at classifying categorical variables based upon their attributes. With the dataset-defined “type” classification as a reference, our 5-class predictive model has shown that LCA as a process can be used to successfully define where relevant classes exist when not otherwise defined from collected data, even with our relatively small dataset (101 points). LCA does have limited scope in that both the indicator(categorical dependent variables) and latent variables must be categorical, where other separate processes such as Latent Profile Analysis (LPA), Latent Trait Analysis (LTA), and Factor Analysis exist for circumstances when one or both of the indicator and latent variables are continuous <span class="citation" data-cites="law2016primer">(<a href="#ref-law2016primer" role="doc-biblioref">Law and Harrington 2016</a>)</span>.</p>
<p>With larger datasets, and proper screening of models, LCA is a powerful tool for finding group commonalities between members of a group, particularly in disciplines where research is often done through self-reporting and assessment of softer-science categories such as opinion, behavioral observations, and medical research<span class="citation" data-cites="kaplan2004sage">(<a href="#ref-kaplan2004sage" role="doc-biblioref">Kaplan 2004</a>)</span>. Even highly complex data can potentially become much more meaningful through the use of Latent Class Analysis, but the substantial challenges that exist with determining the fit of the ideal model require consistent acknowledgment of the possibility for error on the part of researcher, and transparency with regard to the processes used within the research<span class="citation" data-cites="law2016primer">(<a href="#ref-law2016primer" role="doc-biblioref">Law and Harrington 2016</a>)</span>.</p>

</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-asparouhov2014auxiliary" class="csl-entry" role="doc-biblioentry">
Asparouhov, Tihomir, and Bengt Muthén. 2014. <span>“Auxiliary Variables in Mixture Modeling: Three-Step Approaches Using m Plus.”</span> <em>Structural Equation Modeling: A Multidisciplinary Journal</em> 21 (3): 329–41.
</div>
<div id="ref-grimm2021model" class="csl-entry" role="doc-biblioentry">
Grimm, Kevin J, Russell Houpt, and Danielle Rodgers. 2021. <span>“Model Fit and Comparison in Finite Mixture Models: A Review and a Novel Approach.”</span> In <em>Frontiers in Education</em>, 6:613645. Frontiers Media SA.
</div>
<div id="ref-kaplan2004sage" class="csl-entry" role="doc-biblioentry">
Kaplan, Magidson, D. 2004. <em>The SAGE Handbook of Quantitative Methodology for the Social Sciences</em>. SAGE Publications, Inc. <a href="https://methods.sagepub.com/book/the-sage-handbook-of-quantitative-methodology-for-the-social-sciences">https://methods.sagepub.com/book/the-sage-handbook-of-quantitative-methodology-for-the-social-sciences</a>.
</div>
<div id="ref-law2016primer" class="csl-entry" role="doc-biblioentry">
Law, Ernest H, and Rachel Harrington. 2016. <span>“A Primer on Latent Class Analysis.”</span> <em>Value &amp; Outcomes Spotlight</em> 2 (6): 18–19.
</div>
<div id="ref-naldi2020research" class="csl-entry" role="doc-biblioentry">
Naldi, Luigi, and Simone Cazzaniga. 2020. <span>“Research Techniques Made Simple: Latent Class Analysis.”</span> <em>Journal of Investigative Dermatology</em> 140 (9): 1676–80.
</div>
<div id="ref-nylund2018ten" class="csl-entry" role="doc-biblioentry">
Nylund-Gibson, Karen, and Andrew Young Choi. 2018. <span>“Ten Frequently Asked Questions about Latent Class Analysis.”</span> <em>Translational Issues in Psychological Science</em> 4 (4): 440.
</div>
<div id="ref-petersen2019application" class="csl-entry" role="doc-biblioentry">
Petersen, Kimberly J, Pamela Qualter, and Neil Humphrey. 2019. <span>“The Application of Latent Class Analysis for Investigating Population Child Mental Health: A Systematic Review.”</span> <em>Frontiers in Psychology</em> 10: 1214.
</div>
<div id="ref-sinha2021practitioner" class="csl-entry" role="doc-biblioentry">
Sinha, Pratik, Carolyn S Calfee, and Kevin L Delucchi. 2021. <span>“Practitioner’s Guide to Latent Class Analysis: Methodological Considerations and Common Pitfalls.”</span> <em>Critical Care Medicine</em> 49 (1): e63.
</div>
<div id="ref-thorpe2011patterns" class="csl-entry" role="doc-biblioentry">
Thorpe, Joshua M, Carolyn T Thorpe, Korey A Kennelty, and Nancy Pandhi. 2011. <span>“Patterns of Perceived Barriers to Medical Care in Older Adults: A Latent Class Analysis.”</span> <em>BMC Health Services Research</em> 11: 1–12.
</div>
<div id="ref-vermunt2010latent" class="csl-entry" role="doc-biblioentry">
Vermunt, Jeroen K. 2010. <span>“Latent Class Modeling with Covariates: Two Improved Three-Step Approaches.”</span> <em>Political Analysis</em> 18 (4): 450–69.
</div>
<div id="ref-weller2020latent" class="csl-entry" role="doc-biblioentry">
Weller, Bridget E, Natasha K Bowen, and Sarah J Faubert. 2020. <span>“Latent Class Analysis: A Guide to Best Practice.”</span> <em>Journal of Black Psychology</em> 46 (4): 287–311.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>